src/zmq/zmqpublishnotifier.cpp

```diff
@@ -107,6 +110,27 @@ bool CZMQAbstractPublishNotifier::Initialize(void *pcontext)
             return false;
         }
 
+        int enable_ipv6 {1};
+        #if defined(__OpenBSD__)
+        // On OpenBSD the IPv6 option must not be set, if the address to bind isn't IPv6
+        const std::string tcpPrefix = "tcp://";
+        const int tcpIndex = address.rfind(tcpPrefix);
+        const int portColonIndex = address.rfind(":");
+        if (tcpIndex != -1 && portColonIndex != -1) {
+            const std::string ip = address.substr(tcpPrefix.length(), portColonIndex - tcpPrefix.length());
+            CNetAddr addr;
+            LookupHost(ip, addr, false);
+            if (!addr.IsIPv6())
+                enable_ipv6 = {0};
+        }
```

Comment:
This is pretty ugly. Can we just ignore errors setting `ZMQ_IPV6` if the subsequent bind succeeds?

Reply:
I cleaned up a little bit, but I'm not sure, if ignoring an error while setting `ZMQ_IPV6` is a good idea: If it fails, the binding of an IPv6 address will also fail and and prior error could help narrowing down the issue. And if we set the option on e.g. OpenBSD without checking, the bind will always fail for IPv4.
Did I got you right? You want to always set `ZMQ_IPV6` and see if the binding succeeds. If it does, all fine. If it didn't, what then?